cartridges/app_custom_cartridge/cartridge/models/productLineItem/decorators/index.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+module.exports = {
+    gift: require('*/cartridge/models/productLineItem/decorators/gift'),
```

Comment:
You don't need to copy all this from base. Call the base function then only call your function here.